trucks.js

```diff
@@ -1,256 +1,267 @@
-    type: 'Gastropub',
-    schedule: ['Monday', 'Wednesday', 'Friday']
-  },
-  {
-    name: 'Buddha Bruddah',
-    type: 'Asian',
-    schedule: ['Monday', 'Tuesday','Wednesday','Thursday', 'Friday']
-  },
-  {
-    name: 'Budha Bear Bagels',
-    type: 'Sandwiches',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Friday']
-  },
-  {
-    name: 'Bumbu Truck',
-    type: 'Indonesian',
-    schedule: ['Sunday']
-  },
-  {
-    name: 'BUNS',
-    type: 'Burgers',
-    schedule: ['Tuesday', 'Wednesday', 'Friday']
-  },
-  {
-    name: 'Cake Mobile',
-    type: 'Sweets',
-    schedule: []
-  },
-  {
-    name: 'Caravan Crepes',
-    type: 'French',
-    schedule: ['Wednesday', 'Thursday', 'Friday']
-  },
-  {
-    name: 'Charlie\'s',
-    type: 'Burgers/Cheesesteaks',
-    schedule: ['Monday', 'Tuesday', 'Wednesday','Thursday']
-  },
-  {
-    name: 'Cheese Wizards',
-    type: 'Sandwiches',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday']
-  },
-  {
-    name: 'Chewaya Authentic Moroccan Sanwiches',
-    type: 'Moroccan',
-    schedule: []
-  },
-  {
-    name: 'Chick\'n Fix',
-    type: 'American/Fillipino',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday']
-  },
-  {
-    name: 'Chopstix',
-    type: 'Asian Fusion',
-    schedule: ['Tuesday', 'Wednesday', 'Thursday']
-  },
-  {
-    name: 'Contigo',
-    type: 'Mexican',
-    schedule: []
-  },
-  {
-    name: 'Convoy Coffee',
-    type: 'Coffee',
-    schedule: ['Wednesday', 'Saturday', 'Sunday']
-  },
-  {
-    name: 'Crisp Creperie',
-    type: 'French',
-    schedule: ['Tuesday']
-  },
-  {
-    name: 'Curb Jumper Street Eats',
-    type: 'Sandwiches/Burgers',
-    schedule: ['Monday', 'Wednesday']
-  },
-  {
-    name: 'Curbside',
-    type: 'Vietnamese',
-    schedule: []
-  },
-  {
-    name: 'Danielle\'s Crepes',
-    type: 'French',
-    schedule: ['Tuesday', 'Friday']
-  },
-  {
-    name: 'Dante\'s Inferno Dogs',
-    type: 'Hot Dogs',
-    schedule: []
-  },
-  {
-    name: 'Delicatessen Montanti',
-    type: 'Sandwiches',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday']
-  },
-  {
-    name: 'Diablo Food Truckz',
-    type: 'Asian',
-    schedule: []
-  },
-  {
-    name: 'Djung on Wheels',
-    type: 'Thai',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Friday']
-  },
-  {
-    name: 'Dog Japon',
-    type: 'Hot Dogs',
-    schedule: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
-  },
-  {
-    name: 'Dogfather Catering',
-    type: 'Hot dogs, sausages, wood fired pizza',
-    schedule: []
-  },
-  {
-    name: 'Don Lucho\'s',
-    type: 'Peruvian Sandwiches',
-    schedule: []
+  function filterTrucksByDay(day) {
+    function truckFilter(item) {
+      var scheduleLen = item.schedule.length;
+      for (var i=0; i<scheduleLen; i++) {
+        if(item.schedule[i] == day) {
+          return true;
+        }
+      }
+    }
+
+    return (foodTrucks.filter(truckFilter, day));
```

Comment:
Thank you, James. I got your point. Do you post your answers online so I
can see how you did it?

Fulin

On Tue, May 3, 2016 at 11:59 PM, jamdon <notifications@github.com> wrote:

> In trucks.js
> <https://github.com/JSCRIPT300-Spring2016/Module_3/pull/5#discussion_r61995783>
> :
>
> > -  },
> > -  {
> > -    name: 'Don Lucho\'s',
> > -    type: 'Peruvian Sandwiches',
> > -    schedule: []
> > +  function filterTrucksByDay(day) {
> > +    function truckFilter(item) {
> > +      var scheduleLen = item.schedule.length;
> > +      for (var i=0; i<scheduleLen; i++) {
> > +        if(item.schedule[i] == day) {
> > +          return true;
> > +        }
> > +      }
> > +    }
> > +
> > +    return (foodTrucks.filter(truckFilter, day));
>
> This works fine, but you don't need to pass that second parameter, day,
> as you already have access to the variable that was passed in to the outer
> function, filterTrucksByDay through the magic of closures. Additionally,
> it's better to avoid declaring functions inside functions like this.
> Instead, just declare truckFilter inside your trucks module as another
> private function, then use it inside filterTrucksByDay.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/JSCRIPT300-Spring2016/Module_3/pull/5/files/8f59614ffda1b2f9d7266ece22f9cf4aa907bb56#r61995783>
>



-- 
Fulin Shen
541-357-8618